kingpin/actors/base.py

```diff
@@ -90,19 +102,42 @@ def __init__(self, desc, options, dry=False, warn_on_failure=False,
             warn_on_failure: (Bool) Whether this actor ignores its return
                              value and always succeeds (but warns).
             condition: (Bool) Whether to run this actor.
+            init_context: (Dict) Key/Value pairs used at instantiation
+                time to replace {KEY} strings in the actor definition.
+                This is usually driven by the group.Sync/Async actors.
         """
         self._type = '%s.%s' % (self.__module__, self.__class__.__name__)
-        self._desc = desc
         self._options = options
         self._dry = dry
         self._warn_on_failure = warn_on_failure
         self._condition = condition
+        self._init_context = init_context
+
+        # Patch our actor description. This needs to happen separately from the
+        # _fill_in_context() method because it has to happen before we setup
+        # our logger object.
+        try:
+            self._desc = utils.populate_with_tokens(
+                desc, init_context,
+                self.left_context_separator,
+                self.right_context_separator)
+        except LookupError as e:
+            raise exceptions.InvalidOptions(e)
```

Comment:
This is gonna be weird/broken. If it raises - the parent class will try to grab _desc from here, and it'll be empty.

I recommend adding some (sigh) context to the exception message -- such as `Failed to apply context to description: %s`